lib/components/map/itinerary-summary-overlay.tsx

```diff
@@ -0,0 +1,218 @@
+  visible?: boolean
+  visibleItinerary?: number
+}
+
+const Card = styled.div`
+  ${boxShadowCss}
+
+  background: #fffffffa;
+  border-radius: 5px;
+  padding: 5px;
+  align-items: center;
+  display: flex;
+  flex-wrap: wrap;
+
+  
+  span {
+    span {
+      span {
+        max-height: 28px;
+        min-height: 20px;
+      }
+    }
+  }
+  div {
+    margin-top: -0px!important;
+  }
+  .route-block-wrapper span {
+    padding: 0px;
+  }
+  * {
+    height: 27px;
+  }
+}
+`
+
+function addItinLineString(itin: Itinerary): ItinWithGeometry {
+  return {
+    ...itin,
+    allLegGeometry: lineString(
+      itin.legs.flatMap((leg) => polyline.decode(leg.legGeometry.points))
+    )
+  }
+}
+
+type ItinUniquePoint = {
+  itin: Itinerary
+  uniquePoint: Position
+}
+
+function getUniquePoint(
+  thisItin: ItinWithGeometry,
+  otherPoints: ItinUniquePoint[]
+): ItinUniquePoint {
+  const otherMidpoints = otherPoints.map((mp) => mp.uniquePoint)
+  let maxDistance = -Infinity
+  const line = thisItin.allLegGeometry
+  const centerOfLine = centroid(line).geometry.coordinates
+  let uniquePoint = centerOfLine
+
+  line.geometry.coordinates.forEach((point) => {
+    const totalDistance = otherMidpoints.reduce(
+      (prev, cur) => (prev += distance(point, cur)),
+      0
+    )
+
+    const selfDistance = distance(point, centerOfLine)
+    // maximize distance from all other points while minimizing distance to center of our own line
+    const averageDistance = totalDistance / otherMidpoints.length - selfDistance
+
+    if (averageDistance > maxDistance) {
+      maxDistance = averageDistance
+      uniquePoint = point
+    }
+  })
+  return { itin: thisItin, uniquePoint }
+}
+
+const ItinerarySummaryOverlay = ({
+  itins,
+  setActiveItinerary: setActive,
+  setVisibleItinerary: setVisible,
+  visible,
+  visibleItinerary
+}: Props) => {
+  // @ts-expect-error React context is populated dynamically
+  const { LegIcon } = useContext(ComponentContext)
+
+  let sharedTimeout: null | NodeJS.Timeout = null
+
+  if (!itins || !visible) return <></>
+  const mergedItins: ItinWithGeometry[] =
+    doMergeItineraries(itins).mergedItineraries.map(addItinLineString)
+  const midPoints = mergedItins.reduce<ItinUniquePoint[]>((prev, curItin) => {
+    prev.push(getUniquePoint(curItin, prev))
+    return prev
+  }, [])
+  // The first point is probably not well placed, so let's run the algorithm again
+  if (midPoints.length > 1) {
+    midPoints[0] = getUniquePoint(mergedItins[0], midPoints)
+  }
+
+  try {
+    return (
+      <>
+        {midPoints.map(
+          (mp, index) =>
+            // If no itinerary is hovered, show all of them. If one is selected, show only that one
+            // TODO: does this cause an issue with the merging? is the right itinerary selected?`
+            (visibleItinerary !== null && visibleItinerary !== undefined
+              ? visibleItinerary === index
+              : true) &&
+            mp.uniquePoint && (
```

Comment:
These conditions could go into a `midPoints.filter`.

Reply:
They were moved here because of some index matching bug. I agree this entire file could probably be cleaned up, but I'll wait for us to all agree on the algorithm before doing cleanup. I'll add a todo